blmath/geometry/primitives/plane.py

```diff
@@ -262,103 +264,129 @@ def mesh_xsections(self, m, neighborhood=None):
+                # http://code.activestate.com/recipes/498243-finding-eulerian-path-in-undirected-graph/
+                # Under PSF License
+                # NB: MUTATES d
+
+                # counting the number of vertices with odd degree
+                odd = [x for x in self.d if len(self.d[x])&1]
+                odd.append(self.d.keys()[0])
+                if not allow_multiple_connected_components and len(odd) > 3:
+                    return None
+                stack = [odd[0]]
+                path = []
+                # main algorithm
+                while stack:
+                    v = stack[-1]
+                    if v in self.d:
+                        u = self.d[v].pop()
+                        stack.append(u)
+                        self.remove_edge(u, v)
+                    else:
+                        path.append(stack.pop())
+                return path
 
         # 4: Build the edge adjacency graph
         G = Graph(verts.shape[0])
-        def indexof(v, in_this):
-            return np.nonzero(np.all(np.abs(in_this - v) < eps, axis=1))[0]
-        for ii, v in enumerate(verts):
-            for other_v in list(v2s[indexof(v, v1s)]) + list(v1s[indexof(v, v2s)]):
-                neighbors = indexof(other_v, verts)
-                for jj in neighbors:
-                    G.add_edge(ii, jj)
-
-        def euler_path(graph):
-            # Based on code from Przemek Drochomirecki, Krakow, 5 Nov 2006
-            # http://code.activestate.com/recipes/498243-finding-eulerian-path-in-undirected-graph/
-            # Under PSF License
-            # NB: MUTATES graph
-
-            # counting the number of vertices with odd degree
-            odd = [x for x in graph.keys() if len(graph[x])&1]
-            odd.append(graph.keys()[0])
-            # This check is appropriate if there is a single connected component.
-            # Since we're willing to take away one connected component per call,
-            # we skip this check.
-            # if len(odd)>3:
-            #     return None
-            stack = [odd[0]]
-            path = []
-            # main algorithm
-            while stack:
-                v = stack[-1]
-                if v in graph:
-                    u = graph[v].pop()
-                    stack.append(u)
-                    # deleting edge u-v (v-u already removed by pop)
-                    graph[u].remove(v)
-                    # graph[v].remove(u)
-                    if len(graph[v]) == 0:
-                        del graph[v]
-                    if len(graph[u]) == 0:
-                        del graph[u]
-                else:
-                    path.append(stack.pop())
-            return path
+        for f in fs:
+            e0 = intersection_map.index(f[0], f[1])
+            e1 = intersection_map.index(f[0], f[2])
+            e2 = intersection_map.index(f[1], f[2])
+            if e0 is None:
```

Comment:
This is operating on a mathematical certainty that exactly two of the edges of face that intersects the plane will themselves intersect the plane. Can you make that explicit with a comment? The reasoning for this logic otherwise isn't immediately obvious from just the code.